scripts/leetcode_directory_md.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+
+from urllib.request import Request, urlopen
+from pathlib import Path
+from os import listdir
+import json
+from dataclasses import dataclass
+
+
+@dataclass
+class Task:
+    id: str
+    title: str
+    solution: str
+    difficulty: str
+
+def fetch_leetcode_folder_tasks(solutions_folder: Path) -> list[Task]:
+    # Fetch tasks info from the leetcode API.
+    req = Request(
+        url="https://leetcode.com/api/problems/algorithms/", 
+        headers={"User-Agent": "Mozilla/5.0 (X11; U; Linux i686) Gecko/20071127 Firefox/2.0.0.11"}

```

Comment:
You don't need the `User-Agent` header - you are already using the leetcode `api`, they expect that bots or applications would be using this, there isn't any need to make your requests mimic a browser. Use the `requests` library which makes this unnecessary 😄

Reply:
Totally agree, but unfortunally, if there is no "User-Agent" , the server answers "Forbidden". It would be nice to find how to fix this.

Reply:
Use the `requests.get()` you won't need to add the `user-agent` 😄